loaders/key.go

```diff
@@ -19,5 +20,9 @@ type FSKeyLoader struct {
 
 // Load keys from embedded FS
 func (m FSKeyLoader) Load(id circuits.CircuitID) ([]byte, error) {
-	return os.ReadFile(fmt.Sprintf("%s/%v.json", m.Dir, id))
+	file, err := os.ReadFile(fmt.Sprintf("%s/%v.json", m.Dir, id))
```

Comment:
Possible attack: https://rowin.dev/blog/preventing-path-traversal-attacks-in-go
In your case, the path and filename are constants. It is secure. But make sure that the security issue is not reproduced in your implementation.